src/sagemaker_containers/c/libchangehostname.c

```diff
@@ -0,0 +1,68 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the 'License'). You
+// may not use this file except in compliance with the License. A copy of
+// the License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the 'license' file accompanying this file. This file is
+// distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+// ANY KIND, either express or implied. See the License for the specific
+// language governing permissions and limitations under the License.
+
+#include <Python.h>
+#include <stdlib.h>
+
+
+int libchangehostname(char *name, size_t len)
+{
+  const char *val = getenv("SM_CURRENT_HOST");
```

Comment:
i don't really like that it depends on an environment variable that might now be initialized yet.
I understand that right now we call it at 'the right' time, but since it's not really guaranteed by anyone I am worried that it might cause issues in the future.
Also sagemaker-containers is not just a library used by us, we also advertise and help customers to use it in BYOC use case.

Anyway we can switch to use json file or guarantee that it's called only after this environment variable is initialized?